WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardViewModel.swift

```diff
@@ -311,18 +311,24 @@ private extension BlazeCampaignDashboardViewModel {
             self?.updateResults()
         }
         blazeCampaignResultsController.onDidResetContent = { [weak self] in
-            self?.updateResults()
+            self?.refetchAllResultsControllers()
         }
 
         productResultsController.onDidChangeContent = { [weak self] in
             self?.updateAvailability()
             self?.updateResults()
         }
         productResultsController.onDidResetContent = { [weak self] in
+            self?.refetchAllResultsControllers()
             self?.updateAvailability()
-            self?.updateResults()
         }
 
+        refetchAllResultsControllers()
+    }
+
+    /// Refetching all the results controllers is necessary after a storage reset in `onDidResetContent`
+    /// callback and before reloading UI that involves more than one results controller.
+    func refetchAllResultsControllers() {
```

Comment:
💬 How about trying the new notification fix first, to isolate the issue? If the crash still happens with a different stack trace, we know it's a different issue and can consider adding the changes to refetch both results controllers.

Reply:
Sounds good! I revert the change in 361f079f2c0130cfc63afd3ccd03e8829d0e7433.